Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -645,6 +645,58 @@ static void logProcessPoolState(const WebProcessPool& pool)
 {
     m_weakObserver = adoptNS([[WKProcessPoolWeakObserver alloc] initWithWeakPtr:*this]);
 
+#if ENABLE(NOTIFYD_BLOCKING_IN_WEBCONTENT)
+    constexpr ASCIILiteral notificationMessages[] = {
+        "com.apple.WebKit.LibraryPathDiagnostics"_s,
+        "com.apple.WebKit.dumpUntrackedMallocs"_s,
+        "org.WebKit.memoryWarning"_s,
+        "org.WebKit.memoryWarning.begin"_s,
+        "org.WebKit.memoryWarning.end"_s,
+        "org.WebKit.lowMemory"_s,
+        "org.WebKit.lowMemory.begin"_s,
+        "org.WebKit.lowMemory.end"_s,
+        "com.apple.WebKit.fullGC"_s,
+        "com.apple.WebKit.deleteAllCode"_s,
+        "com.apple.WebKit.dumpGCHeap"_s,
+        "com.apple.WebKit.showBackForwardCache"_s,
+        "com.apple.WebKit.showMemoryCache"_s,
+        "com.apple.WebKit.showRenderTree"_s,
+        "com.apple.WebKit.showLayerTree"_s,
+        "com.apple.WebKit.showGraphicsLayerTree"_s,
+        "com.apple.WebKit.showPaintOrderTree"_s,
+        "com.apple.WebKit.showLayoutTree"_s,
+        "com.apple.WebKit.showAllDocuments"_s,
+        "com.apple.WebKit.logMemStats"_s,
+        "com.apple.WebKit.logPageState"_s
+    };
+    m_notifyTokens = WTF::map(notificationMessages, [this](const ASCIILiteral& message) -> int {
+        int notifyToken;
+        notify_register_dispatch(message, &notifyToken, dispatch_get_main_queue(), ^(int token) {
```

Comment:
`notify_register_dispatch` may fail. Maybe we should revise this code slightly to return a `std::optional<int>`, and return `nullopt` it the call fails.

Then we can switch from `WTF::map` to `WTF::compactMap`, which will return the `Vector` of all non-nullopt items.

(@cdumez Is my memory on how `compactMap` works right here?

Reply:
Correct, if notify_register_dispatch() can fail, then we could switch to WTF::compactMap() and have the lambda return a `std::optional<int>`. Then you'd return std::nullopt in case of failure.